src/Server/RestRouterHandlers/RestRouterHandler.h

```diff
@@ -36,30 +36,36 @@ class RestRouterHandler : private boost::noncopyable
     /// a correct `http_status` code will be set by trying best.
     /// This function may throw, and caller will need catch the exception
     /// and sends back HTTP `500` to clients
-    String execute(HTTPServerRequest & request, HTTPServerResponse & response, Int32 & http_status)
+    void execute(HTTPServerRequest & request, HTTPServerResponse & response)
     {
         setupHTTPContext(request);
 
-        http_status = HTTPResponse::HTTP_OK;
+        Int32 http_status = HTTPResponse::HTTP_OK;
+        String response_payload;
 
-        if (streaming())
+        if (streamingInput())
         {
-            return execute(request.getStream(), response, http_status);
+            response_payload = execute(request.getStream(), http_status);
         }
         else
         {
-            String data = "{}";
-
-            auto size = request.getContentLength();
-            if (size > 0)
+            auto payload{parseRequest(request)};
+            if (streamingOutput())
             {
-                data.resize(size);
-                request.getStream().readStrict(data.data(), size);
+                /// For keep-alive to work.
+                if (request.getVersion() == HTTPServerRequest::HTTP_1_1)
+                    response.setChunkedTransferEncoding(true);
+                execute(payload, response);
             }
+            else
+                response_payload = execute(payload, http_status);
+        }
 
-            Poco::JSON::Parser parser;
-            auto payload = parser.parse(data).extract<Poco::JSON::Object::Ptr>();
-            return execute(payload, http_status);
+        /// Only send response_payload when streamingOutput is false, otherwise it will cause output messed up when compressison is enabled.
```

Comment:
```
/// When `streamingOutput` is true, all of the response has already handled
/// We shall not send any other data any more, otherwise the response stream
/// can be corrupted 
```